pavelib/assets.py

```diff
@@ -20,75 +20,174 @@
+        applicable_directories.extend(
+            get_system_sass_dirs(system)
+        )
+
     return applicable_directories
 
 
+def get_theme_sass_dirs(system, theme_dir):
+    """
+    Return list of sass dirs that need to be compiled for the given theme. Also add theme sass directories to
+    THEME_SASS_WATCHER_DIRS variables so that they could be added to sass watcher later.
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    :param theme_dir: absolute path of theme for which to compile sass files.
+    """
+    if system not in ('lms', 'cms'):
+        raise ValueError('"system" must either be "lms" or "cms"')
+
+    dirs = []
+
+    system_sass = path(system) / "static" / "sass"
+    sass = theme_dir / system / "static" / "sass"
+    css = theme_dir / system / "static" / "css"
+
+    if sass.isdir():
+        css.mkdir_p()
+
+        # first compile lms sass files and place css in theme dir
+        dirs.append({
+            "sass_source_dir": system_sass,
+            "css_destination_dir": css,
+            "lookup_paths": [
+                sass / "partials",
+                system_sass / "partials",
+                system_sass,
+            ],
+        })
+
+        # now compile theme sass files and override css files generated from lms
+        dirs.append({
+            "sass_source_dir": sass,
+            "css_destination_dir": css,
+            "lookup_paths": [
+                sass / "partials",
+                system_sass / "partials",
+                system_sass,
+            ],
+        })
+
+        # Add theme dirs to sass watcher directories
+        THEME_SASS_WATCHER_DIRS.extend([sass, sass / 'partials'])
+
+    return dirs
+
+
+def get_system_sass_dirs(system):
+    """
+    Return list of sass dirs that need to be compiled for the given system.
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    """
+    if system not in ('lms', 'cms'):
+        raise ValueError('"system" must either be "lms" or "cms"')
+
+    dirs = []
+    sass = path(system) / "static" / "sass"
+    css = path(system) / "static" / "css"
+
+    dependencies = SASS_LOOKUP_DEPENDENCIES.get(system, [])
+    dirs.append({
+        "sass_source_dir": sass,
+        "css_destination_dir": css,
+        "lookup_paths": dependencies + [
+            sass / "partials",
+            sass,
+        ],
+    })
+
+    if system == 'lms':
+        dirs.append({
+            "sass_source_dir": path(system) / "static" / "certificates" / "sass",
+            "css_destination_dir": path(system) / "static" / "certificates" / "css",
+            "lookup_paths": [
+                sass / "partials",
+                sass
+            ],
+        })
+
+    return dirs
+
+
+def get_watcher_dirs():
+    """
+    :return dirs that need to be added to sass watchers.
+    """
+    dirs = []
+    dirs.extend(COMMON_LOOKUP_DIRS)
+    dirs.extend(THEME_SASS_WATCHER_DIRS)
```

Comment:
I'm incorrect about this, because the calls below pass ``None`` to the theme, so don't update ``THEME_SASS_WATCHER_DIRS``. The only place that this list is updated is ``_compile_sass``, and it isn't clear to me that this will be called before ``get_watcher_dirs`` is called. 

This code is very confusing to understand because of the side effects in various methods. IMO the code needs to have an explicit loop here through all the installed themes to collect the directories to watch, rather than relying upon various side effects updating ``THEME_SASS_WATCHER_DIRS``. In fact, I would remove this global variable entirely because it is very confusing that it is named as if it is a configured constant, but then it is updated via side effect.